pygmt/src/solar.py

```diff
@@ -0,0 +1,92 @@
+"""
+solar - Plot day-night terminators.
+"""
+
+import datetime
+
+import pandas as pd
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import build_arg_string, fmt_docstring, kwargs_to_strings, use_alias
+
+
+@fmt_docstring
+@use_alias(
+    R="region",
+    J="projection",
+    G="fill",
+    B="frame",
+    W="pen",
+    U="timestamp",
+    V="verbose",
+    X="xshift",
+    Y="yshift",
+    p="perspective",
+)
+@kwargs_to_strings(R="sequence", p="sequence")
+def solar(self, terminator="d", terminator_datetime="", **kwargs):
+    r"""
+    Plot day-light terminators.
+
+    This function plots the day-night terminator. Additionally, it can
+    plot the terminators for civil twilight, nautical twilight, and
+    astronomical twilight.
+
+    {aliases}
+
+    Parameters
+    ----------
+    terminator : str
+        Set the type of terminator displayed. Valid arguments are
+        **day_night**, **civil**, **nautical**, and **astronomical**, which
+        can be set with either the full name or the first letter of the name.
+        [Default is ``day_night``]
+    terminator_datetime : str or datetime object
+        Set the UTC date and time of the displayed terminator. It can be
+        passed as a string, using the format *YYYY-MM-DD HH:MM:SS*
+        (e.g. 0600 on January 1, 2000 would be ``2000-01-01 06:00:00``). A
+        datetime object can be passed as well.
+        [Default is the current date and time]
+    {R}
+    {J}
+    {B}
+    fill : str
+        Color or pattern for filling of terminators.
+    pen : str
+        Set pen attributes for lines. The default pen
+        is ``default,black,solid``.
+    {U}
+    {V}
+    {XY}
+    {c}
+    {p}
+    {t}
+    """
+
+    kwargs = self._preprocess(**kwargs)  # pylint: disable=protected-access
+    if "T" not in kwargs:
+        if terminator not in [
+            "day_night",
+            "nautical",
+            "civil",
+            "astronomical",
+            "astro",
+            "d",
+            "n",
+            "c",
+            "a",
+        ]:
+            raise GMTInvalidInput(
+                f"Unrecognized solar terminator type '{terminator}'. Valid values "
+                "are 'day_night', 'civil', 'nautical', and 'astronomical'."
+            )
+        if not terminator_datetime:
+            terminator_datetime = datetime.datetime.now()
+        try:
+            terminator_timestamp = pd.to_datetime(terminator_datetime)
+            datetime_string = terminator_timestamp.strftime("%Y-%m-%dT%H:%M:%S")
+        except ValueError as verr:
+            raise GMTInvalidInput("Unrecognized datetime format.") from verr
+        kwargs["T"] = terminator[0] + "+d" + datetime_string
```

Comment:
We don't set the **T** alias, so there is no need to check `if "T" not in kwargs:` (i.e., using `T="d"` is invalid).

Reply:
Since many of the modules leave the option open for sometime to pass arguments using the GMT format (E= prior to dcw in coast is the first one that comes to mind) I wanted to leave open the open for someone to set "T=" for solar.

Reply:
See #256 for why we want to disable single-letter parameters like `T="d"`, and what we may do (#262).

Reply:
Sounds good. Would it be a good idea to raise a warning if someone does pass a T argument in?

Reply:
@seisman What is a good way to test this  The current test compares the arguments for `terminator` and `terminator_datetime` with the string passed to `T`. Since those two parameters do Python functions instead of just passing a string to the GMT API, I'm assuming we want to cover them in our tests? I'm not sure how to do that if we don't use the `if` statement to see if `T` is in `kwargs`.

Reply:
I opened issue #963 for discussion about similar problems.

Reply:
@seisman What's the best way to wrap this up? My vote would be to leave the current `if` statement and tests as is, and update the testing when we come to a consensus in #963.

Reply:
OK to me.

Reply:
Sounds good. I'll mark this PR ready for review.